### PR TITLE
Transfer a group to another user

### PR DESCRIPTION
**Pull request by khcr**
A button was added in the card's edit page (users/cards).
Once the button is clicked, a pop up displays an explanation text and a field to enter an email. The user can then send the form and see a flash (with either a success or a failure message).

### CODE REVIEW

**Comment by khcr** [comment]
@nkcr please design the transfer button.

File: app/controllers/users/cards_controller.rb [hunk 1]
```diff
@@ -15,6 +15,17 @@ def update
...
   end
 
+  # transfer a card to a new user
+  def transfer
+    user = User.find_by_email(params[:email])
+    unless user.nil?
+      card.update_attribute(:user, user)
+      redirect_to profile_path, error: "Ton groupe a été transféré à l'utilisateur demandé"
+    else
+      redirect_to edit_users_card_path, error: "Aucun utilisateur avec cette email n'a été trouvé"
```
Comment (nkcr) [inline app/controllers/users/cards_controller.rb:25]:
```suggestion
      redirect_to edit_users_card_path, error: "Aucun utilisateur avec l'email #{params[:email]} n'a été trouvé. Sois sûr que la personne ait déjà créé un compte sur le site avant de transférer le groupe."
```

File: app/views/users/cards/edit.html.erb [hunk 2]
```diff
@@ -116,4 +116,14 @@
...
 <!-- end form -->
-<% end %>
\ No newline at end of file
+<% end %>
+
+<!-- contain another form, which means it cannot be nested in the card's edit form -->
+<%= button_to_confirmation "Transférer le groupe", form: { url: transfer_users_card_path(@card) } do %>
+
+  <p>Tu peux transférer la responsabilité du groupe à un autre utilisateur.</p>
+  <p>Indique l'email d'un utilisateur enregistré sur Réseau jeunesse:</p>
```
Comment (nkcr) [inline app/views/users/cards/edit.html.erb:125]:
```suggestion
  <p>Tu peux transférer la responsabilité du groupe à un autre utilisateur. Avant cela sois sûr que la personne à qui tu veux transférer la responsabilité du groupe a créé un compte sur le site. Sinon, demande-lui d'abord de s'inscrire avec ce lien: <%= link_to signup_path, signup_path %></p>
  <p>Indique l'email d'un utilisateur enregistré sur Réseau jeunesse:</p>
```

**Review by nkcr — COMMENTED** [review]
Thanks! 👍 

I think we are missing the email notification part: 
- notify the target user that someone made him responsible of a new card 
- notify the old responsible that the ownership was transferred 

Maybe for later: add a confirmation mechanism. But not sure if that's useful, especially considering the extra friction it adds.

**Review by nkcr — APPROVED** [review]

**Pull request merged**